src/Components/SysTable/hooks.js

```diff
@@ -0,0 +1,135 @@
+import { useApolloClient } from '@apollo/client';
+import { useEffect, useLayoutEffect } from 'react';
+import { useDispatch } from 'react-redux';
+
+export const useInventoryUtilities = (
+    inventory
+) => {
+    const dispatch = useDispatch();
+
+    // Resets the Inventory to a loading state
+    // and prevents previously shown columns and rows to appear
+    useLayoutEffect(() => {
+        dispatch({
+            type: 'INVENTORY_INIT'
+        });
+    }, []);
+
+    // Filters do not yet trigger the inventory to call getEntities
+    // and the page would not reset to page 1
+    const resetPage = () => {
+        Promise.resolve(
+            dispatch({
+                type: 'RESET_PAGE'
+            })
+        ).then(() => inventory?.current?.onRefreshData());
+    };
+
+    useEffect(() => {
+        resetPage();
+    }, []);
+};
+
+export const buildApiFilters = (filters = {}) => {
+    const { tagFilters, ...otherFilters } = filters;
+    const tagsApiFilter = tagFilters
+        ? {
+            tags: tagFilters.flatMap((tagFilter) =>
+                tagFilter.values.map(
+                    (tag) =>
+                        `${encodeURIComponent(tagFilter.key)}/${encodeURIComponent(
+                            tag.tagKey
+                        )}=${encodeURIComponent(tag.value)}`
+                )
+            )
+        }
+        : {};
+
+    return {
+        ...otherFilters,
+        ...tagsApiFilter
+    };
+};
+
+const renameInventoryAttributes = ({
+    culledTimestamp,
+    staleWarningTimestamp,
+    staleTimestamp,
+    insightsId,
+    lastScanned,
+    ...system
+}) => ({
+    ...system,
+    updated: lastScanned,
+    culled_timestamp: culledTimestamp,
+    stale_warning_timestamp: staleWarningTimestamp,
+    stale_timestamp: staleTimestamp,
+    insights_id: insightsId
+});
+
+export const profilesRulesFailed = (profiles) =>
+    profiles.flatMap(
+        (profile) =>
+            profile.rules && profile.rules.filter((rule) => !rule.compliant)
+    );
+
+export const systemRulesFailed = (system) =>
+    system.testResultProfiles
+        ? profilesRulesFailed(system.testResultProfiles)
+        : [];
+
+export const systemSupportedByProfiles = (profiles = []) =>
+    profiles.reduce((acc, profile) => acc && !!profile.supported, true);
```

Comment:
Can be simplified by using destruct
```suggestion
    profiles.reduce((acc, { supported }) => acc && !!supported, true);
```